### PR TITLE
Fix: hide Monitors without a patient checkbox filter is not toggling

### LINKED ISSUE #6991: Hide Monitors without a Patient checkbox filter is not toggling state
**Describe the bug**

**To Reproduce**

1. Go to Central Nursing Station of any facility (https://care.ohc.network/facility/0c95c7f0-e1d2-4aff-83fa-933cef60d3a8/cns)
2. Open Settings & Filters
3. Attempt to uncheck "hide monitors without a patient" checkbox

**Expected behavior**
Should be able to toggle the checkbox state

**Screenshots**
image

### PR DESCRIPTION
**Pull request by GokulramGHV**
## Proposed Changes

- Fixes #6991 

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
why is a state required? query params itself holds the state and it worked before that way.

**Comment by GokulramGHV** [comment]
> why is a state required? query params itself holds the state and it worked before that way.

If I don't use a state then the checkbox doesn't get checked or unchecked quickly, there's some noticeable delay when you do so.

**Comment by rithviknishad** [comment]
why is the noticeable delay being caused?

**Comment by GokulramGHV** [comment]
> why is the noticeable delay being caused?

Sorry my bad, the delay was seemed to have caused because of my computer slowing down. 
I tried again and the checking and unchecking the checkbox works without any delay.
I'll update the PR with the suggested changes.

**Comment by GokulramGHV** [comment]
But there's still a noticeable delay when you click the checkbox and the monitors (without patients) appearing and reappearing due to the api call.
Do you want me to add a loading icon for this?

File: src/Components/Facility/CentralNursingStation.tsx [hunk 1]
```diff
@@ -152,15 +152,12 @@ export default function CentralNursingStation({ facilityId }: Props) {
...
+                          ? true
+                          : qParams.bed_is_occupied === "true"
                       }
                       onChange={({ name, value }) => {
-                        if (value) {
-                          updateQuery({ [name]: value });
-                        } else {
-                          removeFilter(name);
-                        }
+                        updateQuery({ [name]: value });
```
Comment (rithviknishad) [inline src/Components/Facility/CentralNursingStation.tsx]: are you sure this is right? we don't want to filter by `bed_is_occupied==False` when unchecked, rather we don't want to perform the filter itself based on `bed_is_occupied` altogether when unchecked right?

**Comment by rithviknishad** [comment]
> But there's still a noticeable delay when you click the checkbox and the monitors (without patients) appearing and reappearing due to the api call.
> Do you want me to add a loading icon for this?

Yup that sounds good! We can show the CARE loading icon when the useQuery's `loading` is true.

**Review by rithviknishad — APPROVED** [review]
@GokulramGHV 

I've pushed the fix (and also shows the loading component when refetching) as this had to be fixed by today evening. cc: @nihal467 

The issue was the query param variable is of type string, but the checkbox expected value of type boolean. Hence the issue.

**Comment by GokulramGHV** [comment]
> I've pushed the fix (and also shows the loading component when refetching) as this had to be fixed by today evening. cc: @nihal467
> 
> The issue was the query param variable is of type string, but the checkbox expected value of type boolean. Hence the issue.

@rithviknishad Thanks for pushing a fix. I'm so sorry for not seeing your comment earlier and fixing it.

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**